easynav_gps_vff/easynav_gps_vff_controller/src/easynav_gps_vff_controller/GpsVffController.cpp

```diff
@@ -0,0 +1,306 @@
+{
+  auto node = get_node();
+  const auto & plugin_name = get_plugin_name();
+
+  node->declare_parameter<float>(plugin_name + ".distance_obstacle_detection", 3.0);
+  node->declare_parameter<float>(plugin_name + ".distance_to_goal", 1.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_x_min", 0.5);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_x_max", 10.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_y_min", -10.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_y_max", 10.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_z_min", 0.10);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_z_max", 1.00);
+  node->declare_parameter<double>(plugin_name + ".max_speed", 0.8);
+  node->declare_parameter<double>(plugin_name + ".max_angular_speed", 1.5);
+
+  node->get_parameter<float>(plugin_name + ".distance_obstacle_detection",
+      distance_obstacle_detection_);
+  node->get_parameter<float>(plugin_name + ".distance_to_goal", distance_to_goal_);
+  node->get_parameter<float>(plugin_name + ".obstacle_detection_x_min", obstacle_detection_x_min_);
+  node->get_parameter<float>(plugin_name + ".obstacle_detection_x_max", obstacle_detection_x_max_);
+  node->get_parameter<float>(plugin_name + ".obstacle_detection_y_min", obstacle_detection_y_min_);
+  node->get_parameter<float>(plugin_name + ".obstacle_detection_y_max", obstacle_detection_y_max_);
+  node->get_parameter<float>(plugin_name + ".obstacle_detection_z_min", obstacle_detection_z_min_);
+  node->get_parameter<float>(plugin_name + ".obstacle_detection_z_max", obstacle_detection_z_max_);
+  node->get_parameter<double>(plugin_name + ".max_speed", max_speed_);
+  node->get_parameter<double>(plugin_name + ".max_angular_speed", max_angular_speed_);
+
+  // Initialize the odometry message
+  cmd_vel_.header.stamp = node->now();
+  cmd_vel_.header.frame_id = "base_link";
+  cmd_vel_.twist.linear.x = 0.0;
+  cmd_vel_.twist.linear.y = 0.0;
+  cmd_vel_.twist.linear.z = 0.0;
+  cmd_vel_.twist.angular.x = 0.0;
+  cmd_vel_.twist.angular.y = 0.0;
+  cmd_vel_.twist.angular.z = 0.0;
+
+  // Create subscriber to IMU data
+  imu_subscriber_ = node->create_subscription<sensor_msgs::msg::Imu>(
+    "/imu/data", rclcpp::SensorDataQoS().reliable(),
+    std::bind(&GpsVffController::imu_callback, this, std::placeholders::_1));
+
+  // Publisher for visualization markers
+  marker_array_pub_ = node->create_publisher<visualization_msgs::msg::MarkerArray>(
+    "/markers_vff", 10);
+
+  return {};
+}
+
+void GpsVffController::imu_callback(const sensor_msgs::msg::Imu::SharedPtr msg)
+{
+  // Extract the yaw angle from the IMU data
+  tf2::Quaternion q(
+    msg->orientation.x, msg->orientation.y,
+    msg->orientation.z, msg->orientation.w);
+  tf2::Matrix3x3 m(q);
+  double roll, pitch, yaw;
+  m.getRPY(roll, pitch, yaw);
+  yaw_ = yaw;
+}
```

Comment:
Isn't this more related to the localization than the control? Maybe it should be moved to the localization package...